src/matchers/to-contain-element.tsx

```diff
@@ -0,0 +1,40 @@
+import { ReactTestInstance } from 'react-test-renderer';
+import {
+  matcherHint,
+  RECEIVED_COLOR as receivedColor,
+} from 'jest-matcher-utils';
+import { checkHostElement, formatElement } from './utils';
+
+export function toContainElement(
+  this: jest.MatcherContext,
+  container: ReactTestInstance,
+  element: ReactTestInstance | null
+) {
+  if (element !== null || !this.isNot) {
```

Comment:
This check seems to be incorrect. 

There should be non conditional check on the container `checkHostElement(container, toContainElement, this)`

And then conditional check on the element `checkHostElement(element, toContainElement, this);`

https://github.com/testing-library/jest-dom/blob/6d659995665076d78b9f874bec75545bfc56e5ab/src/to-contain-element.js#L4

Reply:
I've corrected and provided relevant tests